Architecture/TodoApp/android/core/src/main/java/com/novoda/todoapp/tasks/data/source/InMemoryLocalTaskDataSource.java

```diff
@@ -82,24 +82,16 @@ public void call(Subscriber<? super SyncedData<Task>> subscriber) {
         }).compose(this.<SyncedData<Task>>delay());
     }
 
-    @Override
-    public Observable<Void> deleteAllTasks() {
-        return Observable.create(new Observable.OnSubscribe<Void>() {
-            @Override
-            public void call(Subscriber<? super Void> subscriber) {
-                localTasks = Tasks.empty();
-                subscriber.onCompleted();
-            }
-        }).compose(this.<Void>delay());
-    }
-
     private <T> Observable.Transformer<T, T> delay() {
         return new Observable.Transformer<T, T>() {
             @Override
             public Observable<T> call(Observable<T> observable) {
                 return observable.doOnEach(new Action1<Notification<? super T>>() {
                     @Override
                     public void call(Notification<? super T> t) {
+                        if (t.getKind() == Notification.Kind.OnCompleted) {
```

Comment:
Yes, exactly in a real API call there would be next to no delay between the onNext and onComplete.